### ISSUE TITLE
new windows beta goes deaf after opening task manager

### ISSUE THREAD

**Issue by karlkleinpaste** [Xpra-org/xpra#3628]
Because my life is somehow incomplete without having an open bug here...

The new MSWindows 4.4-r31990 seems very good, much improved over the old 4.3 I had been using. It has one problem that I've noticed:

Opening the task manager renders the shadow server deaf. Once deaf, it's gone. The only solution is to `taskkill /f /pid 1234` and start over. It's fine when restarted...right up until you open TM again, at which point it goes deaf. Once TM is open, there is no reaction to mouse or keyboard. You can see this from the standpoint that hovering the mouse over window titlebar controls like the red background X "go away" button ... doesn't even highlight. The machine doesn't know that the mouse is there, it has become unaware of it.

1. Start Windows shadow server.
2. `xpra attach ssh://espresso:14500` and login as usual.
3. Play around all you like, everything is fine, until...
4. Open TM. (Easiest invocation: Windows-X, T.)
5. Now deaf, and stays that way. Discon/recon does not fix the problem. Shadow server must be killed/restarted.

Only TM has this deafness effect. I've seen nothing else do it.

Windows 4.4-r31990 shadow server.
4.4-10.r31983 on F35.

Related, because I have been restarting and restarting and restarting: Since I have Cygwin/X sshd access, I would dearly love to have an xpra shadow server invocation that would work from the Cygwin bash command line, or from cmd as invoked from Cygwin bash. As matters stand, I can invoke what I think is the proper command (`start xpra shadow --bind-tcp=0.0.0.0:14500,auth=sys,ssl-cert=auto`, taken from `xpra.iss`) which gives the impression of working, but when I connect to it, the connection closes on a complaint of invalid max size. That's an _exciting_ size misperception.
```
2022-09-27 15:12:40,494  keyboard settings: rules=evdev, model=pc105, layout=us
2022-09-27 15:12:40,497  desktop size is 5760x1080:
2022-09-27 15:12:40,497   :0.0 (1023x191 mm - DPI: 143x144) workarea: 5760x1029 at    0x26  
2022-09-27 15:12:40,497     PRO DP-3         1920x1080 at    0x0    (344x193 mm - DPI: 142x142) workarea: 1920x1029 at    0x26
2022-09-27 15:12:40,498     AUO DP-2         1920x1080 at 1920x0    (344x193 mm - DPI: 142x142) workarea: 1920x1080 at 1920x0
2022-09-27 15:12:40,498     PRO DP-1         1920x1080 at 3840x0    (344x193 mm - DPI: 142x142) workarea: 1920x1080 at 3840x0
2022-09-27 15:12:40,599 enabled remote logging
2022-09-27 15:12:40,600 Xpra shadow server version 4.4
2022-09-27 15:12:40,600  running on unknown
2022-09-27 15:12:40,600  remote desktop size is 582009284x32762
2022-09-27 15:12:40,614 SSH EOF on stderr of run-xpra
2022-09-27 15:12:40,619 removing unix domain socket '/run/user/5271/xpra/clients/godiva.kleinpaste.org-338280'
invalid maximum desktop size: 582009284x32762
```

**Comment by totaam** [comment]
> Opening the task manager renders the shadow server deaf.
> (..)
> Only TM has this deafness effect. I've seen nothing else do it.

This is the same problem we have with the lock screen and login screen (#1527): the task manager is privileged and doesn't run on the same screen, so we can't interact with it. It's not a bug, it's a Microsoft OS security feature.

And I am not aware of any API that would tell us when the screen is changed. If there was one, we could (given the correct permissions of course) follow the screen when it changes, allowing us to follow the user to its desktop after he logs in.

> I would dearly love to have an xpra shadow server invocation that would work from the Cygwin bash command line,
> (..)
> I can invoke what I think is the proper command

It is.
> That's an exciting size misperception

This is a very similar issue to the one above.
The cygwin shell doesn't run within an interactive session.
We can only figure it out by detecting those invalid screen dimensions, everything else looks correct and identical to a working desktop. (API calls, etc)
I think we need to run it via psexec: https://github.com/Xpra-org/xpra/issues/1527#issuecomment-765479686 / #1527
Perhaps we can automate this somehow?
It would be best if the `shadow` command could re-execute itself via `paexec` when it finds that this is needed.

> Where does xpra get the idea that the left monitor is both short (not 1080) and misplaced (0x26)?

That's the workarea for this monitor, not the monitor size which is shown just before.
You must have a taskbar or something on this monitor near the top?
Xpra uses this information to tell applications to avoid placing windows there, so we don't overlap with the taskbar / menu / whatever is there.
You should be able to see these same values by running `NativeGUI_info.exe`.

**Comment by totaam** [comment]
> I would dearly love to have an xpra shadow server invocation that would work from the Cygwin bash command line, or from cmd as invoked from Cygwin bash

With the commits above, this works-for-me(tm)
```shell
xpra shadow ssh://winuser@winhost -d ssh
```

We detect when the desktop is missing using `get_desktop_name() is None` and then re-launch via `paexec` to start the xpra shadow server in the correct context: `-i` (interactive) and `-s` (system) - this may not work with all user accounts..

Tested with both msys openssh server and the regular openssh server.
(I can't test cygwin right now because it just refuses to install.. silently too, but it should work similarly to the msys one)

You need an updated MS Windows server build of Xpra, and probably also an updated client - to be safe.

---

As for the original issue to do with task manager, I think we may have to close this as `cantfix`.

**Comment by karlkleinpaste** [comment]
Updated to r31997 on all machines. No joy on manual start of Windows xpra shadow server.

I ssh'd to the host, then
`cd /cygdrive/c/Program Files/Xpra`
`xpra shadow --bind-tcp=0.0.0.0:14500,auth=sys,ssl-cert=auto &`
Then tried to connect with
`xpra attach -d ssh ssh://bjhwpa:14500`
Log from authentication on down:
```
2022-09-28 12:10:24,819 Auth banner: b'Xpra SSH Server\n\r'
2022-09-28 12:10:24,975 Authentication (none) successful!
2022-09-28 12:10:24,976 will try to run xpra from: ['xpra', '$XDG_RUNTIME_DIR/xpra/run-xpra', '/usr/local/bin/xpra', '~/.xpra/run-xpra', 'Xpra_cmd.exe']
2022-09-28 12:10:24,976 paramiko_run_test_command(transport, echo %OS%)
2022-09-28 12:10:24,976 [chan 0] Max packet in: 4096 bytes
2022-09-28 12:10:24,981 [chan 0] Max packet out: 32768 bytes
2022-09-28 12:10:24,981 Secsh channel 0 opened.
2022-09-28 12:10:24,985 [chan run-test:echo %OS%] Sesch channel 0 request ok
2022-09-28 12:10:24,985 exec_command(echo %OS%) returned
2022-09-28 12:10:24,985 exit status is not ready yet, sleeping
2022-09-28 12:10:24,995 exit status is not ready yet, sleeping
2022-09-28 12:10:25,005 exit status is not ready yet, sleeping
2022-09-28 12:10:25,016 exit status is not ready yet, sleeping
2022-09-28 12:10:25,026 exit status is not ready yet, sleeping
2022-09-28 12:10:25,030 [chan run-test:echo %OS%] EOF received (run-test:echo %OS%)
2022-09-28 12:10:25,030 [chan run-test:echo %OS%] EOF sent (run-test:echo %OS%)
2022-09-28 12:10:25,036 exec_command('echo %OS%')=0
2022-09-28 12:10:25,036 exec_command out=['Windows_NT\r\n']
2022-09-28 12:10:25,036 exec_command err=[]
2022-09-28 12:10:25,036 echo %OS%='Windows_NT'
2022-09-28 12:10:25,036 ssh server OS is 'Windows_NT'
2022-09-28 12:10:25,036 paramiko_run_test_command(transport, REG QUERY "HKEY_LOCAL_MACHINE\Software\Xpra" /v InstallPath)
2022-09-28 12:10:25,036 [chan 1] Max packet in: 4096 bytes
2022-09-28 12:10:25,086 [chan 1] Max packet out: 32768 bytes
2022-09-28 12:10:25,086 Secsh channel 1 opened.
2022-09-28 12:10:25,091 [chan run-test:REG QUERY "HKEY_LOCAL_MACHINE\Software\Xpra" /v InstallPath] EOF received (run-test:REG QUERY "HKEY_LOCAL_MACHINE\Software\Xpra" /v InstallPath)
2022-09-28 12:10:25,180 [chan run-test:REG QUERY "HKEY_LOCAL_MACHINE\Software\Xpra" /v InstallPath] EOF sent (run-test:REG QUERY "HKEY_LOCAL_MACHINE\Software\Xpra" /v InstallPath)
Warning: failed to connect:
 connection failed: Channel closed.
2022-09-28 12:10:25,181 Ignoring message for dead channel 1
2022-09-28 12:10:25,189 removing unix domain socket '/run/user/5271/xpra/clients/godiva.kleinpaste.org-85656'
```

**Comment by karlkleinpaste** [comment]
Identical failure, when starting shadow server from the host itself, that is, start xpra, hit the bottom right "shadow" button. The browse window shows the session is active.

**Comment by totaam** [comment]
Oops. Wait, you're connecting to xpra's `tcp` port in `ssh` mode and not the cygwin or MS Windows openssh server! (which is what I had tested last)
Sorry about that. The commit above fixes that. Xpra's builtin ssh server needs to emulate **all** the commands that xpra paramiko clients use to do their OS and installation path detection.

Note on the subject: these improvements don't work (yet?) for clients that use `--ssh=ssh` or `--ssh=putty.exe` because I can't think of a command line syntax that would be valid for both `cmd.exe` and posix shells. Teaching the xpra client to talk to the `ssh` or `putty` subprocess using pipes would be quite tedious and error prone.

> I ssh'd to the host, then..

FYI: You should no longer need to ssh first. As per my previous comment, this all in one should do it:
```shell
xpra shadow ssh://winuser@winhost -d ssh
```

Assuming that you do need to run the MS Windows shadow server by hand, try using `Xpra_cmd.exe -d ssh ...` so you will get the log output in your terminal if running from the desktop itself.
If running from an ssh shell, `Xpra_cmd.exe` will re-exec itself and the output will go to the usual  `%APPDATA%\Xpra` location.

**Comment by totaam** [comment]
Edit: missed 122cce44c78d5fc91330c769df7242807b2e5f16 from the last comment, this one is in r31999 and later.

**Comment by karlkleinpaste** [comment]
Regrets for not having understood that the `xpra shadow` command was all-in-one.
So, having installed r32000 on the Windows box, and ensuring that F35 was up to date (r31998),
`xpra shadow ssh://host -d ssh` rewards me with ... confusion over whether it thinks xpra is installed at all.

```
2022-09-29 17:24:44,444 Authentication (publickey) successful!
2022-09-29 17:24:44,445 authenticated using agent and key 'MD5:87:b4:ad:51:e8:75:69:76:fe:4f:7e:25:4e:03:70:02'
2022-09-29 17:24:44,445 will try to run xpra from: ['xpra', '$XDG_RUNTIME_DIR/xpra/run-xpra', '/usr/local/bin/xpra', '~/.xpra/run-xpra', 'Xpra_cmd.exe']
2022-09-29 17:24:44,445 paramiko_run_test_command(transport, echo %OS%)
2022-09-29 17:24:44,445 [chan 0] Max packet in: 4096 bytes
2022-09-29 17:24:44,462 Received global request "hostkeys-00@openssh.com"
2022-09-29 17:24:44,462 Rejecting "hostkeys-00@openssh.com" global request from server.
2022-09-29 17:24:44,463 Debug msg: b'/home/francine/.ssh/authorized_keys:4: key options: agent-forwarding port-forwarding pty user-rc x11-forwarding'
2022-09-29 17:24:44,632 [chan 0] Max packet out: 32768 bytes
2022-09-29 17:24:44,632 Secsh channel 0 opened.
2022-09-29 17:24:44,659 [chan run-test:echo %OS%] Sesch channel 0 request ok
2022-09-29 17:24:44,659 exec_command(echo %OS%) returned
2022-09-29 17:24:44,659 exit status is not ready yet, sleeping
2022-09-29 17:24:44,669 exit status is not ready yet, sleeping
2022-09-29 17:24:44,679 exit status is not ready yet, sleeping
2022-09-29 17:24:44,689 exit status is not ready yet, sleeping
2022-09-29 17:24:44,699 exit status is not ready yet, sleeping
2022-09-29 17:24:44,709 exit status is not ready yet, sleeping
2022-09-29 17:24:44,720 exit status is not ready yet, sleeping
2022-09-29 17:24:44,730 exit status is not ready yet, sleeping
2022-09-29 17:24:44,735 [chan run-test:echo %OS%] EOF received (run-test:echo %OS%)
2022-09-29 17:24:44,740 exit status is not ready yet, sleeping
2022-09-29 17:24:44,750 [chan run-test:echo %OS%] EOF sent (run-test:echo %OS%)
2022-09-29 17:24:44,750 exec_command('echo %OS%')=0
2022-09-29 17:24:44,750 exec_command out=['%OS%\n']
2022-09-29 17:24:44,750 exec_command err=[]
2022-09-29 17:24:44,750 echo %OS%='%OS%'
2022-09-29 17:24:44,750 paramiko_run_test_command(transport, echo $OSTYPE)
2022-09-29 17:24:44,750 [chan 1] Max packet in: 4096 bytes
2022-09-29 17:24:44,896 [chan 1] Max packet out: 32768 bytes
2022-09-29 17:24:44,896 Secsh channel 1 opened.
2022-09-29 17:24:44,913 [chan run-test:echo $OSTYPE] Sesch channel 1 request ok
2022-09-29 17:24:44,913 exec_command(echo $OSTYPE) returned
2022-09-29 17:24:44,913 exit status is not ready yet, sleeping
2022-09-29 17:24:44,923 exit status is not ready yet, sleeping
2022-09-29 17:24:44,933 exit status is not ready yet, sleeping
2022-09-29 17:24:44,944 exit status is not ready yet, sleeping
2022-09-29 17:24:44,948 [chan run-test:echo $OSTYPE] EOF received (run-test:echo $OSTYPE)
2022-09-29 17:24:44,954 exit status is not ready yet, sleeping
2022-09-29 17:24:44,964 exit status is not ready yet, sleeping
2022-09-29 17:24:44,969 [chan run-test:echo $OSTYPE] EOF sent (run-test:echo $OSTYPE)
2022-09-29 17:24:44,974 exec_command('echo $OSTYPE')=0
2022-09-29 17:24:44,974 exec_command out=['cygwin\n']
2022-09-29 17:24:44,974 exec_command err=[]
2022-09-29 17:24:44,974 OSTYPE='cygwin'
2022-09-29 17:24:44,974 ssh server OS is 'cygwin'
2022-09-29 17:24:44,974 paramiko_run_test_command(transport, command)
2022-09-29 17:24:44,974 [chan 2] Max packet in: 4096 bytes
2022-09-29 17:24:45,040 [chan 2] Max packet out: 32768 bytes
2022-09-29 17:24:45,040 Secsh channel 2 opened.
2022-09-29 17:24:45,057 [chan run-test:command] Sesch channel 2 request ok
2022-09-29 17:24:45,057 exec_command(command) returned
2022-09-29 17:24:45,057 exit status is not ready yet, sleeping
2022-09-29 17:24:45,067 exit status is not ready yet, sleeping
2022-09-29 17:24:45,078 exit status is not ready yet, sleeping
2022-09-29 17:24:45,088 exit status is not ready yet, sleeping
2022-09-29 17:24:45,095 [chan run-test:command] EOF received (run-test:command)
2022-09-29 17:24:45,098 exit status is not ready yet, sleeping
2022-09-29 17:24:45,108 exit status is not ready yet, sleeping
2022-09-29 17:24:45,110 [chan run-test:command] EOF sent (run-test:command)
2022-09-29 17:24:45,118 exec_command('command')=0
2022-09-29 17:24:45,118 exec_command out=[]
2022-09-29 17:24:45,118 exec_command err=[]
2022-09-29 17:24:45,118 paramiko_run_test_command(transport, command -v xpra)
2022-09-29 17:24:45,119 [chan 3] Max packet in: 4096 bytes
2022-09-29 17:24:45,185 [chan 3] Max packet out: 32768 bytes
2022-09-29 17:24:45,186 Secsh channel 3 opened.
2022-09-29 17:24:45,265 [chan run-test:command -v xpra] Sesch channel 3 request ok
2022-09-29 17:24:45,265 exec_command(command -v xpra) returned
2022-09-29 17:24:45,265 exit status is not ready yet, sleeping
2022-09-29 17:24:45,275 exit status is not ready yet, sleeping
2022-09-29 17:24:45,286 exit status is not ready yet, sleeping
2022-09-29 17:24:45,292 [chan run-test:command -v xpra] EOF received (run-test:command -v xpra)
2022-09-29 17:24:45,296 exit status is not ready yet, sleeping
2022-09-29 17:24:45,306 exit status is not ready yet, sleeping
2022-09-29 17:24:45,316 exit status is not ready yet, sleeping
2022-09-29 17:24:45,318 [chan run-test:command -v xpra] EOF sent (run-test:command -v xpra)
2022-09-29 17:24:45,326 exec_command('command -v xpra')=1
2022-09-29 17:24:45,326 exec_command out=[]
2022-09-29 17:24:45,326 exec_command err=[]
2022-09-29 17:24:45,327 paramiko_run_test_command(transport, command -v $XDG_RUNTIME_DIR/xpra/run-xpra)
2022-09-29 17:24:45,327 [chan 4] Max packet in: 4096 bytes
2022-09-29 17:24:45,372 [chan 4] Max packet out: 32768 bytes
2022-09-29 17:24:45,372 Secsh channel 4 opened.
2022-09-29 17:24:45,389 [chan run-test:command -v $XDG_RUNTIME_DIR/xpra/run-xpra] Sesch channel 4 request ok
2022-09-29 17:24:45,389 exec_command(command -v $XDG_RUNTIME_DIR/xpra/run-xpra) returned
2022-09-29 17:24:45,389 exit status is not ready yet, sleeping
2022-09-29 17:24:45,399 exit status is not ready yet, sleeping
2022-09-29 17:24:45,409 exit status is not ready yet, sleeping
2022-09-29 17:24:45,419 exit status is not ready yet, sleeping
2022-09-29 17:24:45,424 [chan run-test:command -v $XDG_RUNTIME_DIR/xpra/run-xpra] EOF received (run-test:command -v $XDG_RUNTIME_DIR/xpra/run-xpra)
2022-09-29 17:24:45,430 exit status is not ready yet, sleeping
2022-09-29 17:24:45,432 [chan run-test:command -v $XDG_RUNTIME_DIR/xpra/run-xpra] EOF sent (run-test:command -v $XDG_RUNTIME_DIR/xpra/run-xpra)
2022-09-29 17:24:45,440 exec_command('command -v $XDG_RUNTIME_DIR/xpra/run-xpra')=1
2022-09-29 17:24:45,440 exec_command out=[]
2022-09-29 17:24:45,440 exec_command err=[]
2022-09-29 17:24:45,440 paramiko_run_test_command(transport, command -v /usr/local/bin/xpra)
2022-09-29 17:24:45,440 [chan 5] Max packet in: 4096 bytes
2022-09-29 17:24:45,501 [chan 5] Max packet out: 32768 bytes
2022-09-29 17:24:45,501 Secsh channel 5 opened.
2022-09-29 17:24:45,519 [chan run-test:command -v /usr/local/bin/xpra] Sesch channel 5 request ok
2022-09-29 17:24:45,519 exec_command(command -v /usr/local/bin/xpra) returned
2022-09-29 17:24:45,519 exit status is not ready yet, sleeping
2022-09-29 17:24:45,529 exit status is not ready yet, sleeping
2022-09-29 17:24:45,539 exit status is not ready yet, sleeping
2022-09-29 17:24:45,549 exit status is not ready yet, sleeping
2022-09-29 17:24:45,555 [chan run-test:command -v /usr/local/bin/xpra] EOF received (run-test:command -v /usr/local/bin/xpra)
2022-09-29 17:24:45,560 exit status is not ready yet, sleeping
2022-09-29 17:24:45,570 exit status is not ready yet, sleeping
2022-09-29 17:24:45,573 [chan run-test:command -v /usr/local/bin/xpra] EOF sent (run-test:command -v /usr/local/bin/xpra)
2022-09-29 17:24:45,580 exec_command('command -v /usr/local/bin/xpra')=1
2022-09-29 17:24:45,580 exec_command out=[]
2022-09-29 17:24:45,580 exec_command err=[]
2022-09-29 17:24:45,580 paramiko_run_test_command(transport, command -v ~/.xpra/run-xpra)
2022-09-29 17:24:45,580 [chan 6] Max packet in: 4096 bytes
2022-09-29 17:24:45,820 [chan 6] Max packet out: 32768 bytes
2022-09-29 17:24:45,820 Secsh channel 6 opened.
2022-09-29 17:24:45,884 [chan run-test:command -v ~/.xpra/run-xpra] Sesch channel 6 request ok
2022-09-29 17:24:45,884 exec_command(command -v ~/.xpra/run-xpra) returned
2022-09-29 17:24:45,884 exit status is not ready yet, sleeping
2022-09-29 17:24:45,894 exit status is not ready yet, sleeping
2022-09-29 17:24:45,904 exit status is not ready yet, sleeping
2022-09-29 17:24:45,915 exit status is not ready yet, sleeping
2022-09-29 17:24:45,925 exit status is not ready yet, sleeping
2022-09-29 17:24:45,935 exit status is not ready yet, sleeping
2022-09-29 17:24:45,943 [chan run-test:command -v ~/.xpra/run-xpra] EOF received (run-test:command -v ~/.xpra/run-xpra)
2022-09-29 17:24:45,945 exit status is not ready yet, sleeping
2022-09-29 17:24:45,955 exit status is not ready yet, sleeping
2022-09-29 17:24:45,963 [chan run-test:command -v ~/.xpra/run-xpra] EOF sent (run-test:command -v ~/.xpra/run-xpra)
2022-09-29 17:24:45,965 exec_command('command -v ~/.xpra/run-xpra')=1
2022-09-29 17:24:45,966 exec_command out=[]
2022-09-29 17:24:45,966 exec_command err=[]
2022-09-29 17:24:45,966 paramiko_run_test_command(transport, command -v Xpra_cmd.exe)
2022-09-29 17:24:45,966 [chan 7] Max packet in: 4096 bytes
2022-09-29 17:24:46,017 [chan 7] Max packet out: 32768 bytes
2022-09-29 17:24:46,017 Secsh channel 7 opened.
2022-09-29 17:24:46,046 [chan run-test:command -v Xpra_cmd.exe] Sesch channel 7 request ok
2022-09-29 17:24:46,046 exec_command(command -v Xpra_cmd.exe) returned
2022-09-29 17:24:46,046 exit status is not ready yet, sleeping
2022-09-29 17:24:46,056 exit status is not ready yet, sleeping
2022-09-29 17:24:46,066 exit status is not ready yet, sleeping
2022-09-29 17:24:46,076 exit status is not ready yet, sleeping
2022-09-29 17:24:46,086 exit status is not ready yet, sleeping
2022-09-29 17:24:46,097 exit status is not ready yet, sleeping
2022-09-29 17:24:46,101 [chan run-test:command -v Xpra_cmd.exe] EOF received (run-test:command -v Xpra_cmd.exe)
2022-09-29 17:24:46,107 exit status is not ready yet, sleeping
2022-09-29 17:24:46,117 exit status is not ready yet, sleeping
2022-09-29 17:24:46,120 [chan run-test:command -v Xpra_cmd.exe] EOF sent (run-test:command -v Xpra_cmd.exe)
2022-09-29 17:24:46,127 exec_command('command -v Xpra_cmd.exe')=1
2022-09-29 17:24:46,127 exec_command out=[]
2022-09-29 17:24:46,127 exec_command err=[]
xpra initialization error:
 connection failed: all SSH remote proxy commands have failed - is xpra installed on the remote host?
```

**Comment by totaam** [comment]
Cygwin is correctly detected as non-windows:
```
echo %OS%='%OS%'
OSTYPE='cygwin'
ssh server OS is 'cygwin'
```
But then `xpra` is not a valid command for `cygwin` - since it must have been installed using the `EXE` and not a native `cygwin` app:
```
exec_command('command -v xpra')=1
```
So the commit above just treats `cygwin` the same as `msys` and it should now do the same windows registry query to locate the `EXE`.
Builds r32005 and later have this fix. I will have to find a VM to install `cygwin` to test this to be 100% certain, but it should work.

**Comment by totaam** [comment]
So, I installed `cygwin` on a different system.
And of course it wasn't that simple...
First, every cygwin+ssh instruction on the interwebs tells you to run `net start sshd`, but I needed to run `net start cygsshd` instead..
Next, we don't need the same type of string escaping. Because of course not.
And we can't use the same paths either. (`/cygdrive/c/` instead of `/c/`).
And `cygwin` also spawns an ugly shell window that we don't want to see.. etc.
So it does work and maybe it can be improved a bit, but the builtin ssh server from Microsoft or even the one from MSYS2 just work better.
(and now I'm going to have to re-test a lot of ssh connections from / to Linux because these changes were a bit intrusive)

**Comment by karlkleinpaste** [comment]
> Next, we don't need the same type of string escaping. Because of course not.

Let me just say that I appreciate your *ahem* dedication to the cause *cough*.

As for starting Cygwin's sshd, from the directions I posted [in the preceding bug report](https://github.com/Xpra-org/xpra/issues/3626#issuecomment-1257146742), `ssh-host-config` should have started it for you. Hm.

All that said...

**Hallelujah!**

I have a successful xpra connection to a Windows machine for the 1st time in weeks.

And BTW, I don't know what you think you did about things like the Task Manager, which spawned this report originally, but...

**Hallelujah twice!**

...because it no longer goes deaf on TM invocation. I thought your description made this problem out to be insurmountable, but it's operating OK now.

Huge thanx. Closing.
